python/lsst/sims/catUtils/mixins/snUniversalRules.py

```diff
@@ -0,0 +1,171 @@
+import numpy as np
+
+__all__ = ['SNUniverse']
+
+class SNUniverse(object):
+    """
+    Mixin Class providing methods for distributing SN model parameters
+    conditioned  on the parameters of the hostgalaxy from catsim Catalogs
+
+    The base class must have the following attributes:
+    numobjs
+    suppressHighzSN
+    badvalues
+    midSurveyTime
+    suppressDimSN
+    mjdobs
+    maxTimeSNVisible
+
+    """
+
+    @property
+    def averageRate(self):
```

Comment:
You are right. That is a badly named variable. Can we call it SNfrequency and invert it ?